New commands: New-DbaDbFileGroup, Set-DbaDbFileGroup, and Remove-DbaDbFileGroup

lancasteradam opened the pull request:
<!-- Below information IS REQUIRED with every PR -->
## Type of Change
<!-- What type of change does your code introduce -->
 - [x] Bug fix (non-breaking change, fixes #<!--issue number--> )
 - [x] New feature (non-breaking change, adds functionality, fixes #<!--issue number--> )
 - [ ] Breaking change (effects multiple commands or functionality, fixes #<!--issue number--> )
 - [x] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [x] Documentation
 - [ ] Build system
 
### Purpose
- New commands for creating, updating, and removing filegroups.
- Made a minor fix for Disable-DbaFilestream to change RunningValue to [RunValue ](https://docs.microsoft.com/en-us/dotnet/api/microsoft.sqlserver.management.smo.configuration.filestreamaccesslevel)

### Approach
Basic commands for filegroup create, update, remove. All feedback is welcome.

### Commands to test
See the command examples and the integration tests.

potatoqualitee commented:
Thank you, Adam! Will hopefully take a look this weekend.

jpomfret commented on functions/Remove-DbaDbFileGroup.ps1:
```diff
@@ -0,0 +1,106 @@
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$Database,
+        [string]$FileGroupName,
```
Wondering if FileGroup could accept multiple filegroups to be deleted at once?

jpomfret commented on functions/New-DbaDbFileGroup.ps1:
```diff
@@ -0,0 +1,124 @@
+    .PARAMETER Database
+        The target database(s).
+
+    .PARAMETER FileGroupName
```
Could you rename this to just FileGroup for all three functions please? We tend to drop the `name`.

jpomfret reviewed (commented):
This is great work - few ideas\comments for discussion. Thanks for your contributions!

lancasteradam commented:
Thanks @jpomfret, much appreciated. I'll look into those items.

lancasteradam commented:
@jpomfret, thanks again for the feedback. I've incorporated all of the suggestions:

1. Renamed the param for Filegroup.
2. Added support for multiple filegroups names to be passed in for the Set and Remove commands.
3. Added pipeline support for filegroup objects.
4. Added support to set a filegroup as read/write.
5. Added the option to turn off the auto grow of all files.

jpomfret commented on functions/Set-DbaDbFileGroup.ps1:
```diff
@@ -0,0 +1,181 @@
+        [string[]]$Database,
+        [string[]]$FileGroup,
+        [switch]$Default,
+        [switch]$ReadOnly,
+        [switch]$ReadWrite,
```
I think it might be better to handle these as a single param instead of two switches... this currently doesn't fail and seems to set it to ReadWrite.
`Set-DbaDbFileGroup -SqlInstance mssql1 -Database DatabaseAdmin -FileGroup test2New -ReadWrite -ReadOnly`

You could leave it as a switch `$ReadOnly` and people can use `$ReadOnly:$False` to it back to ReadWrite.. or perhaps make it a string that accepts Enabled or Disabled as values?

Tagging @potatoqualitee and @wsmelton - do you have preferences or do we have a standard we should follow here?

jpomfret commented on functions/Set-DbaDbFileGroup.ps1:
```diff
@@ -0,0 +1,181 @@
+        [string[]]$Database,
+        [string[]]$FileGroup,
+        [switch]$Default,
+        [switch]$ReadOnly,
+        [switch]$ReadWrite,
+        [switch]$AutoGrowAllFiles,
+        [switch]$AutoGrowSingleFile,
```
Same for these two, perhaps `$AutoGrowAllFiles` with Enabled\Disabled options?

wsmelton commented:
This needs to be closed and created as PR for each command.

wsmelton commented:
It is easier to review single command than trying to review multiple into one. Believe this has been mentioned before but may be wrong.

jpomfret commented:
Apologies on the close\reopen - miss press on the app

potatoqualitee commented:
These three related commands seem reasonable to me as one PR. I'd prefer making it more convenient for contributors when possible, and allow multi files PRs unless it doesn't make sense for an unforeseen reason 🙇

jpomfret commented:
Makes sense to me @potatoqualitee - thanks.

@lancasteradam - as for the param decisions... I think I'm leaning towards the best option being to make them both switches that when used set the property, and to unset it `$ReadOnly:$false` ... what do you think? Feels like that fits with SSMS having checkboxes?

![image](https://user-images.githubusercontent.com/981370/119229793-6ea3d180-bb11-11eb-8cf8-0905568a6e4d.png)

lancasteradam commented:
@jpomfret, I agree, that sounds good. I'll make those changes.

jpomfret reviewed (approved):
looks good - thanks!

jpomfret commented:
unrelated test failures
